packages/plugins/typescript/react-query/src/fetcher-custom-mapper.ts

```diff
@@ -90,7 +90,9 @@ export class CustomMapperFetcher implements FetcherRenderer {
     this.visitor.reactQueryHookIdentifiersInUse.add(hookConfig.query.hook);
     this.visitor.reactQueryOptionsIdentifiersInUse.add(hookConfig.query.options);
 
-    const options = `options?: ${hookConfig.query.options}<${operationResultType}, TError, TData>`;
+    const options = this.visitor.config.legacyMode
+      ? `options?: ${hookConfig.query.options}<${operationResultType}, TError, TData>`
+      : `options?: Omit<${hookConfig.query.options}<${operationResultType}, TError, TData>, 'queryKey' | 'queryFn' | 'initialData'>`;
```

Comment:
this change implies that you cannot pass `initialData` anymore as `options`. I don't think that is what you intended?

FYI, we added some type narrowing to initialData so that if it is provided, we can assume that `data` is never `undefined`. This is only achievable with overloads, like we do in react-query internally.

However, just using options like before should still work, at least since we fixed the overloads here:
- https://github.com/TanStack/query/pull/3873

which was shipped in 4.0.9

Reply:
@TkDodo 
My pr created with 4.0.4, Omit statement came from  #8116 .
So... We can revert `Omit<xxx, 'queryKey' | 'queryFn' | 'initialData'>` part?

Reply:
There were troubles with the typings, which we since fixed. Ideally, there is no chance necessary now compared to v3